src/decode.rs

```diff
@@ -2251,10 +2254,11 @@ unsafe fn decode_b_inner(
                 }
             }
         }
-        if f.frame_hdr().frame_type.is_inter_or_switch() || f.frame_hdr().allow_intrabc {
+        let frame_hdr = f.frame_hdr();
```

Comment:
It looks like you're redeclaring `frame_hdr` a few times in this function. Was that for borrowing reasons, i.e. to avoid holding a reference when we need to pass `&mut f` to a function?

Reply:
Yeah, I think I've minimized the number of redeclarations of `frame_hdr` that still works with the borrow checker now (in this `fn`).  In some places we pass a `&mut f`, so we need this, and then in other contexts, the `&mut f` is over, so a previous `frame_hdr` works.